lib/oban/queue/executor.ex

```diff
@@ -41,12 +43,25 @@ defmodule Oban.Queue.Executor do
     |> to_module()
     |> apply(:perform, [args, job])
     |> case do
+      :ok ->
```

Comment:
I would love to be able to also return a success tuple in the case that workers are returning data for ease of tests. Anything unexpected would still get caught and would log a warning (which I think is the spirit of the issue), but it allows an escape hatch for the current functionality in a clear way.

One thing off the top of my head that is easier to capture with the output from a worker is what contents were put into a zip file, as it saves us from having to open it to check, or otherwise persist that information.